### PR TITLE
Yog's dolphins with less yet somehow more effort

### PR DESCRIPTION
**Pull request by Bawhoppen**
:cl:
add: Space Dolphins have recently been spotted in the region of space nearby the station.
add: As any Space biologist would know, space dolphins and space carp are natural enemies who will fight to the death over territory, however dolphins will not be inherently hostile to humans, unless provoked of course.
/:cl:

Basically yog's dolphins but with a way way way better sprite (by nienhaus) and some better sounds I ripped from some royalty free place, but without the food shit

### CODE REVIEW

**Comment by Bawhoppen** [comment]
oops onesec

**Comment by kevinz000** [comment]
Noice :D

**Comment by Bawhoppen** [comment]
stop making fun of me I forgot to use the dumb little add command reeee

**Comment by Shadowlight213** [comment]
lmao

**Comment by AnturK** [comment]
Pictures please.

**Comment by Jordie0608** [comment]
You may want to commit this sound file as well.

**Comment by Bawhoppen** [comment]
@Jordie0608 thats a good idea thanks

**Comment by Bawhoppen** [comment]
ss 2016-09-06 at 02 25 49

it's animated ingame similar to carp

**Comment by Jordie0608** [comment]
By the way, where is this sound from?
I edited the soundbite to just the first laugh and I kinda prefer it to the original, see for yourself.
https://dl.dropboxusercontent.com/u/169349932/dolphinassholeedit.ogg
vs
https://dl.dropboxusercontent.com/u/169349932/dolphinasshole.ogg

**Comment by WJohn** [comment]
:+1: Neat sprites.

**Comment by AdamElTablawy** [comment]
these are so much better than the sprites kmc crapped out

**Comment by Super3222** [comment]
Sprites and the sounds Joride0608 posted were pretty good :+1:

**Comment by Bawhoppen** [comment]
@Jordie0608 I kind of like my original a little more because it sounds more retarded but both are good, and this is where its from https://www.youtube.com/watch?v=ewd4AgMGsnA I don't know much about licensing but that should be fine

**Comment by Jordie0608** [comment]
Well the licence in that link is CC BY 3.0, I also found the exact same sound on a royalty-free website; but either way the comment indicates it's actually an edited recording of a kookaburra made for the film Flipper.
So... `¯\_(ツ)_/¯`

**Comment by Bawhoppen** [comment]
I mean either way its not like it would ever matter

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/mob/living/simple_animal/hostile/retaliate/dolphin
+	name = "space dolphin"
+	desc = "Yawn."
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: this needs a better desc.

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	icon = 'icons/mob/broadMobs.dmi'
+	icon_state = "dolp"
+	icon_living = "dolp"
+	icon_dead = "dolp_dead"
+	icon_gib = "dolp_gib"
+	pixel_x = -16
+	mob_size = MOB_SIZE_LARGE
+	speak_chance = 0
+	turns_per_move = 5
+	butcher_results = list(/obj/item/organ/brain = 4) //dolphins are intelligent beings
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: no meat?

Comment (Bawhoppen) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: thats part of the joke (not saying its a good joke)

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/mob/living/simple_animal/hostile/retaliate/dolphin
+	name = "space dolphin"
+	desc = "That's LIEUTENANT commander to you!" //sounds like a reference but is not
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: First of all, don't leave comments like that, and second can we please have at least half of a normal description?

Comment (Bawhoppen) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: okay give me a description that isn't retarded and I will put it in

**Comment by PKPenguin321** [comment]
so space carp are a dwarf fortress reference right

are these also or what am i missing here

**Comment by KorPhaeron** [comment]
I think these are just more space sea animals from newfags who don't know that carp are from DF and are just building on space carp

**Comment by Bawhoppen** [comment]
carp are their own thing don't give me the DF crap

File: code/modules/events/carp_migration.dm [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /datum/round_event_control/carp_migration
 	name = "Carp Migration"
 	typepath = /datum/round_event/carp_migration
-	weight = 15
+	weight = 10
```
Comment (optimumtact) [inline code/modules/events/carp_migration.dm]: Do not include unrelated rebalances in your feature pr unless you can coherently explain why you have included it.

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	speed = 0
+	maxHealth = 60 //as strong as megacarp
+	health = 60
+	a_intent = "harm"
+
+	environment_smash = 0
+	harm_intent_damage = 8
+	melee_damage_lower = 20 //dolphins are bitches
+	melee_damage_upper = 20
+	pass_flags = PASSTABLE
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: Why do they have PASSTABLE

**Comment by AdamElTablawy** [comment]
Story behind space dolphins-

Kmc got bored and decided that the game needed a crucial splash of autism and he decided 'hey lol we have space carp why not space dolphins' and added dolphins, missing the point of, y'know, DF reference and all.

Then he shat out a bad sprite and added a passive mob and boom, Dolphins, who really serve no purpose, float aimlessly in space and didn't look cute-until now.

Later he realized dolphins were kind of boring so he decided to add dolphin meat dishes for the chef (fair enough) and hardsuits made out of dolphin fins and syndicate dolphins.

Yeah.

Of course after the rebase we all had a good laugh about it and never thought of adding something that stupid again oh wait no sorry I mean they're trying to add the fucking dolphin hardsuits again. Kill me.

**Comment by optimumtact** [comment]
dolphin hardsuits lmao

File: code/modules/events/carp_migration.dm (hunk 4 above)
Comment (Bawhoppen) [inline code/modules/events/carp_migration.dm]: if I left carp unchanged and added dolphins with weight 15 then we'd havetwice the chance of carp based events so I split them (approx.)

**Comment by Bawhoppen** [comment]
space sea creatures are their own thing now you can't fight the power

**Comment by A-t48** [comment]
Cybernetic dolphin when?

image

File: code/modules/events/carp_migration.dm (hunk 4 above)
Comment (optimumtact) [inline code/modules/events/carp_migration.dm]: alright, thank you

**Comment by Shadowlight213** [comment]
Rnd is now Research and Dolphins

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm (hunk 5 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: They swim over them

**Comment by Bawhoppen** [comment]
do I have to add the damn food shit before this can get merged?

**Comment by Shadowlight213** [comment]
You also need to add holodolphins

**Comment by KorPhaeron** [comment]
Yes, please add the food

File: code/modules/events/carp_migration.dm (hunk 4 above)
Comment (KorPhaeron) [inline code/modules/events/carp_migration.dm]: So basically you added a 50% chance to make carp non hostile?

Whats the point?

**Comment by Bawhoppen** [comment]
okay give me a few days I am a bit busy right now and I will add the food and mess with the spawning event and shit

**Comment by AdamElTablawy** [comment]
don't forget your **dolphin hardsuits**

**Comment by KorPhaeron** [comment]
stale, let me know when you finish

**Comment by Bawhoppen** [comment]
i forgot about this, will finish very soon, reopen if you would

**Comment by francinum** [comment]
O L D P R

**Comment by Bawhoppen** [comment]
goddamn git ruins me every time okay anyways I've set the weight for carp to be 14 which is slightly less than it used to be and the weight for dolp to be 4 which is quite small, and I've added the food but I'd like to get some WJ sprites

**Comment by Iamgoofball** [comment]
@Bawhoppen are you seriously this retarded that carp isn't a DF joke?

**Comment by optimumtact** [comment]
ping to reopen when you're ready to fix conflicts and the grammar issues.

**Comment by Bawhoppen** [comment]
@Iamgoofball ??? 
what does their original reference material matter for? its not like we follow our references to a tee here, our xenos are completely different than the official canon yet we still use them

Spacefauna is its own thing now

**Comment by optimumtact** [comment]
We are trying to stick to a hard science fiction theme where possible and people do not seem to like this very much either.

Apologies.

**Comment by optimumtact** [comment]
@tgstation/commit-access whatever someone else decide on this I'm sick of baw's whining.

**Comment by Bawhoppen** [comment]
just wait for me to get sprites then merge itll be okay

**Comment by Bawhoppen** [comment]
don't close this I am going to give it some !!UNIQUE FEATURES!!

**Comment by Bawhoppen** [comment]
Okay, in the interest of adding unique mechanics, I have made it so dolphins will now drop their fin, which when crafted in conjunction with some bluespace crystals, metal, and wires, can make a radical mini-thruster, of which can then be attached to a scooterframe to make a gnarly Space surfboard (just a slower speedbike)

File: code/game/objects/items/weapons/miscellaneous.dm [hunk 6]
```diff
@@ -17,3 +17,22 @@
...
+	desc = "Could possibly make a good subspace rudder."
+	name = "space dolphin fin"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "dolpfin"
+	force = 6
+	w_class = 3
+	attack_verb = list("slapped")
+
+/obj/item/weapon/dolphinfinthruster
+	desc = "A dolphin fin with some attached miniature bluespace thrusters. Ripe to be assembled into something."
```
Comment (PKPenguin321) [inline code/game/objects/items/weapons/miscellaneous.dm]: Le bluespace technobabble

**Comment by PKPenguin321** [comment]
I do like space surfboards though

**Comment by Bawhoppen** [comment]
I am going to add one other thing too and I'm still waiting on some sprites
Also I realise that I need to fix something

**Comment by Iamgoofball** [comment]
now it's redeemable

**Comment by Ninja137** [comment]
But why

**Comment by ChangelingRain** [comment]
I don't like dolphins you should make megacarp less rare instead

**Comment by Bawhoppen** [comment]
dont close i just need some more time i am busy

**Comment by Bawhoppen** [comment]
Just a little longer I swear also why all the downthumbs

**Comment by Kmc2000** [comment]
Can confirm, sprites were shat out


Also...dolphin hardsuits....? Is it good that i don't remember those?

**Comment by Bawhoppen** [comment]
I'm going to get back to this by the way

**Pull request closed without merging**